pkg/redshift/driver/rows.go

```diff
@@ -201,6 +201,7 @@ func (r *Rows) fetchNextPage(token *string) error {
 func convertRow(columns []*redshiftdataapiservice.ColumnMetadata, data []*redshiftdataapiservice.Field, ret []driver.Value) error {
 	for i, curr := range data {
 		if curr.IsNull != nil && *curr.IsNull {
+			ret[i] = nil
```

Comment:
not setting the `ret`urned value for a certain index was what caused the previous value to be used.

Reply:
wow, looks like this was quite difficult to catch!